src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -348,40 +472,55 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 								  "(OIDS) option in their definitions.")));
 	}
 
-	/* verify target relation is either regular or foreign table */
-	relationKind = relation->rd_rel->relkind;
-	if (relationKind != RELKIND_RELATION && relationKind != RELKIND_FOREIGN_TABLE)
-	{
-		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
-						errmsg("cannot distribute relation: %s",
-							   relationName),
-						errdetail("Distributed relations must be regular or "
-								  "foreign tables.")));
-	}
-
-#if (PG_VERSION_NUM >= 100000)
-	if (relation->rd_rel->relispartition)
-	{
-		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
-						errmsg("cannot distribute relation: %s", relationName),
-						errdetail("Distributing partition tables is unsupported.")));
-	}
-
+	/* verify target relation does not use identity columns */
 	if (RelationUsesIdentityColumns(relationDesc))
 	{
 		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 						errmsg("cannot distribute relation: %s", relationName),
 						errdetail("Distributed relations must not use GENERATED "
 								  "... AS IDENTITY.")));
 	}
-#endif
 
-	/*
-	 * Distribution column returns NULL for reference tables,
-	 * but it is not used below for reference tables.
-	 */
-	distributionColumn = BuildDistributionKeyFromColumnName(relation,
-															distributionColumnName);
+	if (PartitionedTable(relationId))
```

Comment:
MX check is added. There are no partition table check because we prevent distributing partitions directly in the check above.